### PR TITLE
daemon: move signal setup into a function

### PR DESCRIPTION
**Pull request by ngortheone**

### CODE REVIEW

**Comment by ngortheone** [comment]
cc @kevans91 @bsdimp

File: usr.sbin/daemon/daemon.c [hunk 1]
```diff
@@ -431,28 +424,8 @@ main(int argc, char *argv[])
 	state.pipe_fd[1] = -1;
 
 	setproctitle("%s[%d]", state.title, (int)pid);
-	/*
```
Comment (kevans91) [inline usr.sbin/daemon/daemon.c:434]: This doesn't seem to be fundamentally changed by this commit, and AFAICT it's still true; why remove the comments?

File: usr.sbin/daemon/daemon.c [hunk 2]
```diff
@@ -515,6 +488,48 @@ daemon_sleep(time_t secs, long nsecs)
...
+	act_chld.sa_handler = handle_chld;
+	sigemptyset(&act_chld.sa_mask);
+	sigaddset(&act_chld.sa_mask, SIGTERM);
+	if (sigaction(SIGCHLD, &act_chld, NULL) == -1) {
+		warn("sigaction");
+		daemon_terminate(state);
+	}
+
+	/* Setup SIGHUP if configured*/
+	if (state->log_reopen == false || state->output_fd < 0) {
```
Comment (kevans91) [inline usr.sbin/daemon/daemon.c:522]:
```suggestion
	if (!state->log_reopen || state->output_fd < 0) {
```

File: usr.sbin/daemon/daemon.c [hunk 3]
```diff
@@ -312,57 +343,19 @@ main(int argc, char *argv[])
...
-		sigaddset(&mask_susp, SIGCHLD);
-		/* Block SIGTERM when we lack a valid child PID. */
-		sigaddset(&mask_term, SIGTERM);
-		/*
-		 * When reading, we wish to avoid SIGCHLD. SIGTERM
-		 * has to be caught, otherwise we'll be stuck until
-		 * the read() returns - if it returns.
-		 */
-		sigaddset(&mask_read, SIGCHLD);
+		setup_signals(&state);
```
Comment (kevans91) [inline usr.sbin/daemon/daemon.c:346]: Let's move this back under the SIGTERM block; we don't want to create a window where the SIGTERM handler can run before we're ready.

Comment (ngortheone) [reply usr.sbin/daemon/daemon.c]: done

File: usr.sbin/daemon/daemon.c [hunk 4]
```diff
@@ -312,57 +343,20 @@ main(int argc, char *argv[])
...
-		sigaddset(&mask_susp, SIGCHLD);
-		/* Block SIGTERM when we lack a valid child PID. */
-		sigaddset(&mask_term, SIGTERM);
-		/*
-		 * When reading, we wish to avoid SIGCHLD. SIGTERM
-		 * has to be caught, otherwise we'll be stuck until
-		 * the read() returns - if it returns.
-		 */
-		sigaddset(&mask_read, SIGCHLD);
+
```
Comment (kevans91) [inline usr.sbin/daemon/daemon.c:346]: Extra whitespace; kill this line

File: usr.sbin/daemon/daemon.c (hunk 1 above)
Comment (ngortheone) [reply usr.sbin/daemon/daemon.c:434]: I consolidated scattered comments that talk about signal handling nuances in one place, closer to the beginning of the main()

File: usr.sbin/daemon/daemon.c [hunk 5]
```diff
@@ -312,57 +343,20 @@ main(int argc, char *argv[])
...
-			warn("sigaction");
-			daemon_terminate(&state);
-		}
-		if (sigaction(SIGCHLD, &act_chld, NULL) == -1) {
-			warn("sigaction");
-			daemon_terminate(&state);
-		}
+
+		setup_signals(&state);
 		/*
```
Comment (kevans91) [inline usr.sbin/daemon/daemon.c:354]: Newline before multiline comments after the start of the containing block

File: usr.sbin/daemon/daemon.c [hunk 6]
```diff
@@ -431,28 +425,8 @@ main(int argc, char *argv[])
...
-		 * - read until EOF
-		 * - read until EOF but only for a while
-		 * - bail immediately
-		 *
-		 * We go for the third, as otherwise we have no guarantee
-		 * that we won't block indefinitely if the child refuses
-		 * to depart. To handle the second option, a different
-		 * approach would be needed (procctl()?)
-		 */
+
```
Comment (kevans91) [inline usr.sbin/daemon/daemon.c:429]: Extra whitespace; kill this line

File: usr.sbin/daemon/daemon.c [hunk 7]
```diff
@@ -515,6 +489,48 @@ daemon_sleep(time_t secs, long nsecs)
 		}
 	}
 }
+/*
```
Comment (kevans91) [inline usr.sbin/daemon/daemon.c:492]: And a newline here

**Comment by ngortheone** [comment]
merged by @kevans91

**Pull request closed without merging**